### DISCUSSION TITLE
Render grid to PDF

### DISCUSSION THREAD

**Discussion by FraserElectronics** [AvaloniaUI/Avalonia#13358] [Q&A] [answered]
I am trying to render an Avalonia.Controls.DataGrid to PDF. I have found some ideas online and have put together the following code which I think might be close to what I need. However, ImmediateRenderer cannot be found even though it appears to exist in the API documentation (Avalonia.Rendering.ImmediateRenderer).

```
        [RelayCommand]
        public void OnPrintReportButtonClicked( object cmdparam )
        {
            if ( cmdparam is DataGrid grid )
            {
                using var doc = SKDocument.CreatePdf( @"c:\temp\grid.pdf" );
                var page = doc.BeginPage( 5f, 5f );
                using ( var bmp = new SKBitmap() )
                { 
                    using ( var context = DrawingContextHelper.WrapSkiaCanvas( page, SkiaPlatform.DefaultDpi ) )
                    {
                       ImmediateRenderer.Render( this, context );
                    }
                }
                doc.EndPage();
                doc.Close();
            }
        }
```

Can anyone help me?

Andy

**Comment by ykarpeev** [comment]
Check this discussion: https://github.com/AvaloniaUI/Avalonia/discussions/10386

**Reply by FraserElectronics** [reply]
Thanks for the link - very helpful 👍

**Comment by maxkatz6** [comment]
In 11.1 (or nighty), there is a new helper method to render control directly to the skcanvas:
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Skia/Avalonia.Skia/Helpers/DrawingContextHelper.cs#L14

**Reply by FraserElectronics** [reply, marked as answer]
Thanks - that is what I am looking for 👍
